Foundation/String/UTF8.hs

```diff
@@ -872,17 +874,19 @@ data Encoding
     deriving (Typeable, Data, Eq, Ord, Show, Enum, Bounded)
 
 fromEncoderBytes :: ( Encoder.Encoding encoding
-                    , Exception (Encoder.Error encoding)
                     , PrimType (Encoder.Unit encoding)
                     )
                  => encoding
                  -> UArray Word8
                  -> (String, Maybe ValidationFailure, UArray Word8)
 fromEncoderBytes enc bytes =
-    ( String $ runST $ Encoder.convertFromTo enc EncoderUTF8 (Vec.recast bytes)
-    , Nothing
-    , mempty
-    )
+    case runST $ Encoder.convertFromTo enc EncoderUTF8 (Vec.recast bytes) of
+        -- TODO: Don't swallow up specific error (second element of pair)
+        -- TODO: Confused why all this recasting is necessary. I "typed hole"-ed my way to get this function to compile.  Feels like there should be a cleaner method.
```

Comment:
we discussed this, and the original `recast` was a bad idea in the first place. so we'll sort this out with a overall of the decoder that I outlined in #353

In this case instead of recasting the array, you should downcast the `Offset` back to `Word8` unit, so that you can splice the original array in 2 more efficiently.